src/esp/assets/GenericSemanticMeshData.cpp

```diff
@@ -221,33 +212,66 @@ GenericSemanticMeshData::buildSemanticMeshData(
           // color is found in ssd mapping, so is legal color
           semanticID = ssdColorToIDAndRegionIter->second.first;
           regionID = ssdColorToIDAndRegionIter->second.second;
+
         } else {
           // color is not found in ssd mapping, so not legal color
+          // use currently assigned unknown color's semantic ID for this vertex
+          semanticID = nonSSDObjID;
+          regionID = maxRegion;
+
           // check if we've assigned a semantic ID to this color before, and if
           // not do so
-          auto nonSSDClrRes =
-              nonSSDVertColors.insert({meshColorInt, nonSSDObjID});
-          if (nonSSDClrRes.second) {
-            // inserted, so increment nonSSDObjID
-            ++nonSSDObjID;
-          }
-          // map holds that color's nonSSDObjID
-          semanticID = nonSSDClrRes.first->second;
-          regionID = maxRegion;
 
-          // color for given semantic ID - only necessary to add for unknown
-          // colors
-          if (colorMapToUse.size() <= semanticID) {
-            colorMapToUse.resize(semanticID + 1);
+          auto nonSSDClrCountRes =
+              nonSSDVertColorCounts.insert({meshColorInt, 1});
+          if (nonSSDClrCountRes.second) {
+            // unknown color has not been seen before
+            ESP_DEBUG() << dbgMsgPrefix << "Inserted Unknown semantic Color"
+                        << meshColor << "in map w/ nonSSDObjID =" << semanticID;
+            nonSSDVertColorIDs.insert({meshColorInt, nonSSDObjID});
+            // inserted, so increment nonSSDObjID tracking expanded semantic IDs
+            // for future unknown color
+            ++nonSSDObjID;
+            // add color for given semantic ID to map
+            if (colorMapToUse.size() <= semanticID) {
+              colorMapToUse.resize(semanticID + 1);
+            }
+            colorMapToUse[semanticID] = meshColor;
+          } else {
+            ++nonSSDClrCountRes.first->second;
           }
-          colorMapToUse[semanticID] = meshColor;
         }
-        // semantic ID for vertex
+
+        // assign semantic ID for vertex
         semanticData->objectIds_[vertIdx] = semanticID;
         // partition Ids for each vertex, for multi-mesh construction.
         partitionIds[vertIdx] = regionID;
+      }  // for each vertex
+      // FOR VERT-BASED OBB CALC
+      // build semantic OBBs here
+      semanticData->buildSemanticOBBs(semanticData->cpu_vbo_,
+                                      semanticData->objectIds_, ssdObjs,
+                                      dbgMsgPrefix);
+      // colors found on vertices not found in semantic lexicon :
+      if (nonSSDVertColorIDs.size() == 0) {
```

Comment:
```suggestion
      if (nonSSDVertColorIDs.empty()) {
```

Reply:
[you know that's what is being done, right?](https://github.com/gcc-mirror/gcc/blob/9a92e46c0e9a75cd14125493b8826d3e33dd0f67/libstdc%2B%2B-v3/include/bits/hashtable.h#L650) :ferris_wheel: